src/commands/destroy.js

```diff
@@ -26,25 +27,31 @@ export default {
     const room = Room.getById(socket.user.roomId);
     if (type === 'mob') {
       // look for mob in user's current room
-      const mob = autocomplete.autocompleteTypes(socket, ['mob'], name);
-      if (!mob) {
+      const acResult = autocomplete.autocompleteTypes(socket, ['mob'], name);
+      if (!acResult) {
         return;
       }
+      const mob = acResult.item;
 
-      room.mobs.remove(mob);
-      socket.emit('output', { message: 'Mob successfully destroyed.' });
-
-      // announce mob disappearance to any onlookers
-      socket.broadcast.to(room.id).emit('output', { message: 'Mob erased from existence!' });
+      // mobs is a non-mongoose array, so must use removeItem
+      let removedItem = utils.removeItem(room.mobs, mob);
+      if (!removedItem) {
+        socket.emit('output', { message: 'Something went terribly wrong.' });
+      } else {
+        socket.emit('output', { message: 'Mob successfully destroyed.' });
+        // announce mob disappearance to any onlookers
+        socket.broadcast.to(room.id).emit('output', { message: 'Mob erased from existence!' });
+      }
     }
     else if (type === 'item') {
-      const item = autocomplete.autocompleteTypes(socket, ['inventory'], name);
-      if (!item) {
+      const acResult = autocomplete.autocompleteTypes(socket, ['inventory'], name);
+      if (!acResult) {
         return;
       }
 
       // delete item
-      socket.user.inventory.remove(item);
+      // inventory is a mongoose-controlled array, so this must use .remove
+      socket.user.inventory.id(acResult.item.id).remove();
```

Comment:
For arrays that contain EmbeddedObjects (any array that is saved to the database that contains model instances) this is prescribed method of removing an item.